stuff/cron/plagiarism_detector.py

```diff
@@ -0,0 +1,282 @@
+        copydetector_result: List[CopyDetectorResult] = [
+            CopyDetectorResult(
+                test_similarity=test_sim,
+                reference_similarity=ref_sim,
+                test_filename=test_f,
+                reference_filename=ref_f,
+                highlighted_test_code=hl_code_1,
+                highlighted_reference_code=hl_code_2,
+            ) for test_sim, ref_sim, test_f, ref_f, hl_code_1, hl_code_2,
+            overlap in detector.get_copied_code_list()
+        ]
+
+        filter_and_format_result(dbconn, contest_id, submissions,
+                                 copydetector_result)
+
+
+def download_submission_files(dbconn: lib.db.Connection, dir: str,
+                              download: SubmissionDownloader,
+                              submission_ids: Iterable[Submission]) -> None:
+
+    for submission in submission_ids:
+        submission_path = os.path.join(dir, str(submission['problem_id']))
+        os.makedirs(os.path.dirname(submission_path), exist_ok=True)
+        download(submission['guid'], submission_path, submission['language'])
+
+
+def get_contests(dbconn: lib.db.Connection) -> Iterable[Contest]:
+
+    with dbconn.cursor(dictionary=True) as cur:
+        cur.execute(CONTESTS_TO_RUN_PLAGIARISM_ON)
+        return typing.cast(Iterable[Contest], cur.fetchall())
+
+
+def get_submissions_for_contest(dbconn: lib.db.Connection,
+                                contest_id: int) -> Iterable[Submission]:
+
+    with dbconn.cursor(dictionary=True) as cur:
+        cur.execute(GET_CONTEST_SUBMISSION_IDS, (contest_id, ))
+        return typing.cast(Iterable[Submission], cur.fetchall())
+
+
+def run_detector_for_contest(dbconn: lib.db.Connection,
+                             download: SubmissionDownloader,
+                             contest_id: int) -> None:
+
+    with tempfile.TemporaryDirectory(prefix='plagiarism_detector') as dir:
+        submissions = get_submissions_for_contest(dbconn, contest_id)
+        download_submission_files(dbconn, dir, download, submissions)
+        run_copy_detect(dbconn, dir, contest_id, submissions)
+
+
+def main() -> None:
+    ''' Main entrypoint. '''
+    parser = argparse.ArgumentParser(
+        description='Runs the Plagiarism Detector')
+    parser.add_argument('--local-downloader-dir')
+
+    lib.db.configure_parser(parser)
+    lib.logs.configure_parser(parser)
+
+    args = parser.parse_args()
+    args.verbose = True
+    lib.logs.init(parser.prog, args)
+
+    logging.info('started')
+    dbconn = lib.db.connect(lib.db.DatabaseConnectionArguments.from_args(args))
+
+    if args.local_downloader_dir != None:
+        download: SubmissionDownloader = LocalSubmissionDownloader(
+            args.local_downloader_dir)
+    else:
+        download = S3SubmissionDownloader()
+
+    for contest in get_contests(dbconn):
+        run_detector_for_contest(dbconn, download, contest['contest_id'])
+
+    # dbconn.conn.commit()
```

Comment:
```suggestion
    dbconn.conn.commit()
```

we should probably commit this.

Reply:
Done.